cf-atcp-emailer/src/main/java/org/broadinstitute/ddp/cf/ATCPContactEmailer.java

```diff
@@ -0,0 +1,191 @@
+package org.broadinstitute.ddp.cf;
+
+import java.io.IOException;
+import java.nio.channels.Channels;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.functions.BackgroundFunction;
+import com.google.cloud.functions.Context;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.sendgrid.Attachments;
+import com.sendgrid.Email;
+import com.sendgrid.Mail;
+import com.sendgrid.Method;
+import com.sendgrid.Personalization;
+import com.sendgrid.Request;
+import com.sendgrid.Response;
+import com.sendgrid.SendGrid;
+
+public class ATCPContactEmailer implements BackgroundFunction<ATCPContactEmailer.FileScanResult> {
```

Comment:
@YuriyPavlovBI Once again, I made several changes:

* Function now listens for file scan pubsub result instead of bucket notification.
* Use env vars instead of config file.
* More logging and error handling.

Reply:
How do we know if a file was uploaded for DAR form and not as an answer to the file upload question?

Reply:
DAR form uses a separate bucket. The bucket name is passed in as an env var and we match on that when processing the result. We also have that `FILENAME_PATTERN` regex.